Subject: ld: add i386/x86_64-substrate target with new emulations

Default emul is elf_i386_substrate (and elf_x86_64_substrate for 64-bit targets); fall-back emuls cover plain elf_i386 / elf_x86_64 / elf_iamcu so cross-linking object files produced by other Unix toolchains still works.

---
diff --git a/ld/configure.tgt b/ld/configure.tgt
--- a/ld/configure.tgt	2026-05-11 21:33:54.231149445 -0500
+++ b/ld/configure.tgt	2026-05-11 21:36:32.261251599 -0500
@@ -363,6 +363,11 @@
 i[3-7]86-*-redox*)	targ_emul=elf_i386
 			targ_extra_emuls=elf_x86_64
 			;;
+i[3-7]86-*-substrate*)	targ_emul=elf_i386_substrate
+			targ_extra_emuls="elf_i386 elf_iamcu"
+			targ64_extra_emuls="elf_x86_64_substrate elf_x86_64"
+			targ64_extra_libpath="elf_x86_64_substrate elf_x86_64 elf32_x86_64"
+			;;
 i[3-7]86-*-solaris2*)
 			targ_emul=elf_i386_sol2
 			targ_extra_emuls=elf_i386
@@ -965,6 +970,11 @@
 x86_64-*-redox*)	targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386
 			;;
+x86_64-*-substrate*)	targ_emul=elf_x86_64_substrate
+			targ_extra_emuls="elf_x86_64 elf_i386_substrate elf_i386 elf_iamcu"
+			targ_extra_libpath="elf_i386_substrate elf_i386 elf32_x86_64"
+			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/x86_64/i386/'`
+			;;
 x86_64-*-solaris2*)
 			targ_emul=elf_x86_64_sol2
 			targ_extra_libpath=elf_i386_sol2
